R/derive_merged.R

```diff
@@ -347,18 +385,26 @@ derive_vars_merged <- function(dataset,
       select(!!!by_vars_right, !!!replace_values_by_names(new_vars))
   }
 
+  if (!is.null(exist_flag)) {
+    add_data <- mutate(
+      add_data,
+      !!exist_flag := TRUE
+    )
+  }
+
   if (!is.null(missing_values)) {
-    match_flag_var <- get_new_tmp_var(add_data, prefix = "tmp_match_flag")
+    missing_values_var <- get_new_tmp_var(add_data, prefix = "tmp_missing_flag")
   } else {
-    match_flag_var <- match_flag
+    missing_values_var <- missing_values
   }
-
-  if (!is.null(match_flag_var)) {
+  if (!is.null(missing_values_var)) {
     add_data <- mutate(
       add_data,
-      !!match_flag_var := TRUE
+      !!missing_values_var := TRUE
```

Comment:
Why do you create two variables (`!!exist_flag` and `!!missing_values_var`) for the same purpose? Couldn't we use just one?

Reply:
I thought so too and was trying to adopt your `if (is.null(missing_values) || is.null(exist_flag))` idea but I just kept breaking the tests one way or another, seemed to never be able to stich it all together, either the exist flag would be removed, or not properly renamed, etc. 

I think by trying to use the same temporary variable here highlighted below, it would do some funny business, when one or the other argument is missing, never quite able to achieve the functionality highlighted in test 8, so that's why I kept them separate. 

![image](https://github.com/pharmaverse/admiral/assets/43018762/a29daa43-44c1-441a-b6fd-f9da7ea11b69)

Reply:
I would expect that it works with
```
  if (!is.null(missing_values_var)) {
    update_missings <- map2(
      syms(names(missing_values)),
      missing_values,
      ~ expr(if_else(is.na(!!match_flag_var), !!.y, !!.x))
      ~ expr(if_else(is.na(!!missing_values_var), !!.y, !!.x))
    )
    names(update_missings) <- names(missing_values)
    dataset <- dataset %>%
      mutate(!!!update_missings)
  }

  if (!is.null(exist_flag)) {
    dataset <- dataset %>%
      mutate(!!exist_flag := ifelse(is.na(!!missing_values_var), false_value, true_value))
  }

  dataset %>%
      remove_tmp_vars()
```
Or did I miss something?

Reply:
I'm getting a break when `missing_values` is `NULL` but specifying a `exist_flag` column in this scenario

Reply:
Where does it break?

Just for completeness, I would change
```
  if (!is.null(exist_flag)) {
    add_data <- mutate(
      add_data,
      !!exist_flag := TRUE
    )
  }

  if (!is.null(missing_values)) {
    missing_values_var <- get_new_tmp_var(add_data, prefix = "tmp_missing_flag")
  } else {
    missing_values_var <- NULL
  }
```
to
```
  if (!is.null(missing_values) || !is.null(exist_flag)) {
    missing_values_var <- get_new_tmp_var(add_data, prefix = "tmp_missing_flag")
  } else {
    missing_values_var <- NULL
  }
```
and line 431 from
```
  if (!is.null(missing_values_var)) {
```
to
```
  if (!is.null(missing_values)) {
```

Reply:
![image](https://github.com/pharmaverse/admiral/assets/43018762/5c8a5e77-e18a-47ed-8607-513a05168d70)
will cause 
![image](https://github.com/pharmaverse/admiral/assets/43018762/ef19db63-d1b1-4a93-bcad-1561ddb2edc4)
on unit-test 3

Reply:
Strange, I just tried it out and it works for me. I have committed my code. Could you check?

Reply:
Yea looks good now even on the CI/CD, I guess I must've missed something, tough day haha